crates/sui-framework/packages/dwallet-system/sources/dwallet_2pc_mpc_ecdsa_k1.move

```diff
@@ -456,6 +468,37 @@ module dwallet_system::dwallet_2pc_mpc_ecdsa_k1 {
         dwallet_public_key: vector<u8>,
     ): bool;
 
+    /// Encrypt a user share with an AHE encryption key.
+    public fun encrypt_user_share(
+        dwallet: &DWallet<Secp256K1>,
+        encryption_key: &EncryptionKey,
+        encrypted_secret_share_and_proof: vector<u8>,
+        ctx: &mut TxContext,
+    ) {
+        let is_valid = verify_encrypted_user_secret_share(
+            get_encryption_key(encryption_key),
+            encrypted_secret_share_and_proof,
+            get_output(dwallet),
+        );
+
+        assert!(is_valid, EEncryptUserShare);
+
+        create_encrypted_user_share(
+            object::id(dwallet),
+            encrypted_secret_share_and_proof,
+            object::id(encryption_key),
+            ctx
+        );
+    }
+
+    #[allow(unused_function)]
+    native fun verify_encrypted_user_secret_share(
```

Comment:
```suggestion
    native fun verify_encrypted_user_secret_share_secp256k1(
```
I'm not sure, but I think that the native function's should have unique names right, so probably better to have a unique name so it wouldn't cause compilation problems in the future